instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/AttributesExtractor.java

```diff
@@ -45,8 +45,7 @@ protected abstract void onEnd(
    * Sets the {@code value} with the given {@code key} to the {@link AttributesBuilder} if {@code
    * value} is not {@code null}.
    */
-  protected static <T> void set(
-      AttributesBuilder attributes, AttributeKey<T> key, @Nullable T value) {
+  default <T> void set(AttributesBuilder attributes, AttributeKey<T> key, @Nullable T value) {
```

Comment:
Since set isn't really part of the contract of AttributesExtractor, we should keep them off the method even as defaults. I'm thinking the use case of "fill attribute if available" is so common that it's fine to add the `@Nullable` annotation in the API and consider it "correct usage". So easiest is probably to rely on the implementation detail that `put` can be called directly, and I can work on that in the SDK to formalize it.

Can be another PR but we should do it